File: Segundo_Curso/Segundo_Cuatri/PDOO/java/src/irrgarten/Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author el_dramas
 */
public class Shield extends CombatElement {
       
    /**
     * @brief Constructor, construye un escudo nuevo.
     * @param defense Defensa del escudo
     * @param durability Usos que le quedan al escudo
     */
    Shield (float defense, int durability ){
        super(defense,durability);
    }

    Shield(Shield other){
        super(other);
    }

    /**
     * Proteje al jugador que porte el escudo si no está roto,
     * disminuyendo la vida útil del escudo en uno.
     * @return El valor de la absorción de daño del escudo
     */
    public float protect(){
        return super.produceEffect();
    }
    /**
     * @brief Representa la instancia como string
     * @return Los atributos del escudo
     */
    @Override
    public String toString(){
        return "S" + super.toString();
    }

}

File: Segundo_Curso/Segundo_Cuatri/PDOO/ruby/Irrgarten/Shield.rb
#encoding: UTF-8

require_relative 'Dice'
require_relative 'CombatElement'

module Irrgarten
    # Esta clase pretende crear un objeto escudo que se encargará de
    # proteger a los individuos de juego
    # @param protection Cantidad de daño que protege
    # @param uses Cantidad de usos
    class Shield < CombatElement

	# Metodo inicializador de la clase Shield
	# @param defense Proteccion del escudo
	# @param durability Usos del escudo
	# @pos La instancia creada queda inicializada
    # def initialize(defense,durability)
    #     super
    # end
    
	# Efectúa la proteccion de un golpe
	# @return Valor de proteccion
	# pre -> delega en produceEffect
    # see -> CombatElement -> produceEffect
    def protect
        self.produceEffect
    end

	# Se encarga de eliminar el escudo si no hay usos suficientes
	# @return Decisión de inutilizacion
	# @pos Si no hya usos suficientes el objeto quedará inutilizable
	def discard
		Dice.discardElement(@uses)
	end

	# Metodo to string encargado de encadenar el estado del objeto
	# @return cadena con toda la informacion
        def to_s
            "S" + super
        end

    end#class
end#module
